### DISCUSSION TITLE
Create trigger to update profiles.email based on auth.users.email change

### DISCUSSION THREAD

**Discussion by AlexanderCryptoOwl** [supabase/supabase#5421] [Questions] [answered]
Hello there :). 

I understand that this will insert the user into my public.profiles table when a new row is added to auth.users. 

```
-- inserts a row into public.users
create function public.handle_new_user() 
returns trigger 
language plpgsql 
security definer set search_path = public
as $$
begin
  insert into public.profiles (id,email)
  values (new.id, new.email);
  return new;
  
end;
$$;

-- trigger the function every time a user is created
create trigger on_auth_user_created
```

How would I adjust it to make it update an existing user in case they change their email address?.

Was thinking something like this but getting an error 

`"public.profiles" is not a known variable`

```
--update a row at public.users
create function public.handle_update_user() 
returns trigger 
language plpgsql 
security definer set search_path = public
as $$
begin
  update into public.profiles (email)
  values (new.email);
  return new;
  
end;
$$;

-- trigger the function every time a user email is updated
create trigger on_auth_user_updated
  after update on auth.users.email
  for each row execute procedure public.handle_update_user();
```

**Comment by renatocaliari** [comment]
@AlexanderCryptoOwl
It seems maybe the problem is with the UPDATE command using INSERT properties like "INTO" and "VALUES", for example. 

You could try...
- change from this: "update into public.profiles (email) values (new.email); return new;"
- to this: "update public.profiles SET email = new.email; return new;"

I tried both ways here. The first way, like yours, gives the error you mentioned. The second runs successfully.

**Comment by pierre-lgb** [comment, marked as answer]
This should be working with the following changes:

```diff
create or replace function public.handle_update_user() 
returns trigger 
language plpgsql 
security definer set search_path = public
as $$
begin
- update into public.profiles (email)
- values (new.email);
+ update public.profiles
+ set email = new.email
+ where id = new.id;
  return new;
end;
$$;

create trigger on_auth_user_updated
- after update on auth.users.email
+ after update of email on auth.users
  for each row execute procedure public.handle_update_user();
```

You can found the correct usage for SQL `UPDATE` statement [here](https://www.w3schools.com/SQL/sql_update.asp).

Hope this helps!
